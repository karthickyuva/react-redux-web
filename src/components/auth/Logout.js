import React, { Component } from "react";
import { connect } from "react-redux";
import * as actions from "../../actions/auth";
import { Link } from "react-router-dom";

class Logout extends Component {
	componentWillMount() {
		this.props.logoutUser();
	}

	render() {
		return (
			<div>
				<div className="col-md-4 col-md-offset-4 vertical-center text-center col-sm-8 col-sm-offset-2">
					<div className="card" style={{ margin: "0 auto" }}>
						<p style={{ fontSize: "24px" }}>Sorry to see you go</p>
						<p><Link to="/login" className="btn btn-primary">Login</Link></p>
					</div>
				</div>
			</div>
		);
	}
}

export default connect(null, actions)(Logout);
