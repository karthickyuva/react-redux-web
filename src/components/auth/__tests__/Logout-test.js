import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import Logout from "../Logout";
import { MemoryRouter } from "react-router";

function mockStore(state) {
	return {
		"default": () => {},
		subscribe: () => {},
		dispatch: () => {},
		getState: () => state,
		mapDispatchToProps: () => {},
		mapStateToProps: () => {}
	};
}

const store = mockStore({});

describe("Logout", () => {
	it("should display", () => {
		const component = renderer.create(
			<Provider store={store}>
				<MemoryRouter><Logout /></MemoryRouter>
			</Provider>
		);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
