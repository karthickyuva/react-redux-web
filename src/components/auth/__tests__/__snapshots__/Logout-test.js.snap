// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout should display 1`] = `
<div>
  <div
    className="col-md-4 col-md-offset-4 vertical-center text-center col-sm-8 col-sm-offset-2"
  >
    <div
      className="card"
      style={
        Object {
          "margin": "0 auto",
        }
      }
    >
      <p
        style={
          Object {
            "fontSize": "24px",
          }
        }
      >
        Sorry to see you go
      </p>
      <p>
        <a
          className="btn btn-primary"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </p>
    </div>
  </div>
</div>
`;
